app/database/operator/server_data_operator/handlers/custom_profile.ts

```diff
@@ -0,0 +1,82 @@
+import {logWarning} from '@utils/log';
+
+import type ServerDataOperatorBase from '.';
+import type Model from '@nozbe/watermelondb/Model';
+import type {CustomProfileAttribute} from '@typings/api/custom_profile_attributes';
+import type {
+    HandleCustomProfileFieldsArgs,
+    HandleCustomProfileAttributesArgs,
+} from '@typings/database/database';
+
+const {CUSTOM_PROFILE_FIELD, CUSTOM_PROFILE_ATTRIBUTE} = MM_TABLES.SERVER;
+
+export interface CustomProfileHandlerMix {
+    handleCustomProfileFields: ({fields, prepareRecordsOnly}: HandleCustomProfileFieldsArgs) => Promise<Model[]>;
+    handleCustomProfileAttributes: ({attributes, prepareRecordsOnly}: HandleCustomProfileAttributesArgs) => Promise<Model[]>;
+}
+
+const CustomProfileHandler = <TBase extends Constructor<ServerDataOperatorBase>>(superclass: TBase) => class extends superclass {
+    /**
+     * handleCustomProfileFields: Handler responsible for the Create/Update operations occurring on the CUSTOM_PROFILE_FIELD table from the 'Server' schema
+     * @param {HandleCustomProfileFieldsArgs} fieldsArgs
+     * @param {CustomProfileField[]} fieldsArgs.fields
+     * @param {boolean} fieldsArgs.prepareRecordsOnly
+     * @returns {Promise<Model[]>}
+     */
+    handleCustomProfileFields = async ({fields, prepareRecordsOnly = true}: HandleCustomProfileFieldsArgs): Promise<Model[]> => {
+        if (!fields?.length) {
+            logWarning(
+                'An empty or undefined "fields" array has been passed to the handleCustomProfileFields method',
+            );
+            return [];
+        }
+
+        // Cast fields to RawValue[] to satisfy TypeScript
+        const createOrUpdateRawValues = getUniqueRawsBy({raws: fields as any, key: 'id'});
+
+        return this.handleRecords({
+            fieldName: 'id',
+            transformer: transformCustomProfileFieldRecord,
+            createOrUpdateRawValues,
+            tableName: CUSTOM_PROFILE_FIELD,
+            prepareRecordsOnly,
+        }, 'handleCustomProfileFields');
+    };
+
+    /**
+     * handleCustomProfileAttributes: Handler responsible for the Create/Update operations occurring on the CUSTOM_PROFILE_ATTRIBUTE table from the 'Server' schema
+     * @param {HandleCustomProfileAttributesArgs} attributesArgs
+     * @param {CustomProfileAttributeSimple[]} attributesArgs.attributes
+     * @param {boolean} attributesArgs.prepareRecordsOnly
+     * @returns {Promise<Model[]>}
+     */
+    handleCustomProfileAttributes = async ({attributes, prepareRecordsOnly = true}: HandleCustomProfileAttributesArgs): Promise<Model[]> => {
+        if (!attributes?.length) {
+            logWarning(
+                'An empty or undefined "attributes" array has been passed to the handleCustomProfileAttributes method',
+            );
+            return [];
+        }
+
+        const createOrUpdateRawValues = getUniqueRawsBy({raws: attributes, key: 'id'}) as CustomProfileAttribute[];
```

Comment:
The casting should not be needed, in theory. Looking at the code, I see we do the casting in other places probably because of historical reasons, but it is no longer needed.